src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -340,6 +387,73 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    public void validateContentType(FailureCollector failureCollector) {
+      switch (format) {
+        case FORMAT_AVRO:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_AVRO)) {
+            failureCollector.addFailure(String.format("Valid content type for avro is %s",
+                                                      CONTENT_TYPE_APPLICATION_AVRO), null)
+              .withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_JSON:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_JSON)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for json are %s, %s", CONTENT_TYPE_APPLICATION_JSON,
+              CONTENT_TYPE_TEXT_PLAIN),
+                                        null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_CSV:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for csv are %s, %s, %s", CONTENT_TYPE_APPLICATION_CSV,
+              CONTENT_TYPE_TEXT_PLAIN,
+              CONTENT_TYPE_TEXT_CSV),
+                                        null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_DELIMITED:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_TSV)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for delimited are %s, %s, %s, %s", CONTENT_TYPE_TEXT_PLAIN,
+              CONTENT_TYPE_TEXT_CSV, CONTENT_TYPE_APPLICATION_CSV, CONTENT_TYPE_TEXT_TSV), null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_PARQUET:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)) {
+            failureCollector.addFailure(String.format("Valid content type for parquet is %s",
+                                                      CONTENT_TYPE_TEXT_PLAIN),
```

Comment:
Is text/plain the right content type for parquet and orc formats?

Reply:
I changed `text/plain `to `application/octet-stream` for parquet and orc formats. This is also the default content type.